keylime/src/list_parser.rs

```diff
@@ -0,0 +1,297 @@
+                    self.consumed += 1;
+                    match self.parse_name(bracketed, true, '\'') {
+                        Ok(s) => r.push(s),
+                        Err(e) => match e {
+                            Error::EmptyString(_) => {
+                                warn!("{e}")
+                            }
+                            e => {
+                                return Err(e);
+                            }
+                        },
+                    }
+                }
+                _ => match self.parse_name(bracketed, false, ',') {
+                    Ok(s) => r.push(s),
+                    Err(e) => match e {
+                        Error::EmptyString(_) => {
+                            warn!("{e}")
+                        }
+                        e => {
+                            return Err(e);
+                        }
+                    },
+                },
+            }
+        }
+        Ok(r)
+    }
+
+    fn parse_name(
+        &mut self,
+        bracketed: bool,
+        quoted: bool,
+        closer: char,
+    ) -> Result<String, Error> {
+        let mut s = String::new();
+        loop {
+            match self.c.peek() {
+                Some(_) => {
+                    let c = self.c.next().ok_or(Error::EOF)?;
+                    self.consumed += 1;
+
+                    match c {
+                        ']' => {
+                            if bracketed && !quoted {
+                                break;
+                            }
+                            s.push(c);
+                        }
+                        ',' | ' ' | '\n' | '\r' | '\t' => {
+                            // If not quoted, these characters separate names
+                            if !quoted {
+                                break;
+                            }
+
+                            s.push(c);
+                        }
+                        '\"' | '\'' => {
+                            if quoted && closer == c {
+                                break;
+                            }
+                            s.push(c);
+                        }
+                        _ => {
+                            s.push(c);
+                        }
+                    }
+                }
+                None => {
+                    if quoted {
+                        return Err(Error::ClosingQuote(
+                            closer,
+                            self.consumed,
+                        ));
+                    }
+                    if bracketed {
+                        return Err(Error::ClosingBracket(self.consumed));
+                    }
+                    break;
+                }
+            }
+        }
+        if s.is_empty() {
+            Err(Error::EmptyString(self.consumed))
+        } else {
+            Ok(s)
+        }
+    }
+}
+
+pub fn parse_list(list: &str) -> Result<Vec<String>, Error> {
```

Comment:
Is my understanding correct that the type of vector elements can be either a quoted string (`"foo"`) or a symbolic name (`default`)? If so I would define an enum like:
```rust
pub enum ListItem {
  QuotedString(String),
  Name(String),
}
```

Reply:
Sorry, I thought that a typical  input is something like `["/path/to/somewhere", default, ...]`, where the first element is a path while the second element is a symbol with special meaning. But if it treats everything as a string, distinct types wouldn't be needed.